server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightUtils.java

```diff
@@ -43,10 +38,8 @@ public static List<Object> loadFieldValues(
         FetchSubPhase.HitContext hitContext
     ) throws IOException {
         if (fieldType.isStored()) {
-            CustomFieldsVisitor fieldVisitor = new CustomFieldsVisitor(singleton(fieldType.name()), false);
-            hitContext.reader().document(hitContext.docId(), fieldVisitor);
-            List<Object> textsToHighlight = fieldVisitor.fields().get(fieldType.name());
-            return Objects.requireNonNullElse(textsToHighlight, Collections.emptyList());
+            List<Object> values = hitContext.loadedFields().get(fieldType.name());
```

Comment:
To check my understanding: is this the part where we actually save having to  open stored field readers more than once? Does the hitContext now always already contain the loaded fields?

Reply:
Yes, exactly so